enterprise/core-edge/src/test/java/org/neo4j/coreedge/raft/log/segmented/TermsTest.java

```diff
@@ -276,16 +276,35 @@ public void shouldAppendAfterSkip() throws Exception
         assertTermInRange( skipIndex + 1, skipIndex + 20, skipTerm );
     }
 
+    @Test
+    public void shouldMaintainSaneMemoryUseForTerms() throws Exception
```

Comment:
You still need to assert on the actual terms, i.e. which terms are expected to still exist and which are expected to have been pruned out. Would be good to test a few different scenarios, for example pruning nothing, pruning everything, pruning on the boundary of a range, pruning in the middle of a range, pruning a few ranges.